### ISSUE TITLE
Loss of SQUIRREL_TEMP env variable

### ISSUE THREAD

**Issue by tcsidevtools** [clowd/Clowd.Squirrel#136]
I've noticed a potential breaking change from Squirrel.Windows to Clowd.Squirrel. This exists within the Setup project specifically in Setup.cpp

in line 24 we are hardcoding the path squirrel uses to unpack the setup
```
wstring updaterPath = util::get_temp_file_path(L"exe");
```

```
std::wstring util::get_temp_file_path(wstring extension)
{
    wchar_t tempFolderBuf[MAX_PATH];
    DWORD cTempFolder = GetTempPath(MAX_PATH, tempFolderBuf);
    wchar_t tempFileBuf[MAX_PATH];
    GetTempFileName(tempFolderBuf, L"squirrel", 0, tempFileBuf);
    DeleteFile(tempFileBuf);
    wstring tempFile(tempFileBuf);

    if (!extension.empty())
        tempFile += L"." + extension;

    return tempFile;
}
```

This method uses the underlining GetTempPathW from fileapi to do its bidding. And there doesn't appear to be a way to override this behavior. Previously squirrel did this

```
	wchar_t* envSquirrelTemp = _wgetenv(L"SQUIRREL_TEMP");
	if (envSquirrelTemp &&
		DirectoryExists(envSquirrelTemp) &&
		DirectoryIsWritable(envSquirrelTemp) &&
		!PathIsUNCW(envSquirrelTemp)) {
		_swprintf_c(targetDir, _countof(targetDir), L"%s", envSquirrelTemp);
		goto gotADir;
	}
```
and then provided a fallback directory if that environmental variable wasn't set.

This is potentially important for clients with strict antiviruses who do not want to exclude the entire temp directory. So a few questions with this are. 

1. Have I overlooked something and there is a way to override this behavior?
2. If no, Would you accept a PR, which brings the previous behavior back?

I will also mention i recognize that you can override the TMP/TEMP variables. But we wouldnt want to leave those modified. Were looking for something we can set once and have it work for all squirrel installers moving forward.

**Comment by caesay** [comment]
1) You have not overlooked anything, I did remove this. 
2) I could accept a PR but there are some conditions I think for this getting accepted. 

- This needs to be changed in the installer where you pointed out, and also in the C# library here https://github.com/clowd/Clowd.Squirrel/blob/develop/src/Squirrel/Internal/Utility.cs#L270
- The solution needs to use a different environment variable name to Squirrel.Windows, eg. `CLOWD_SQUIRREL_TEMP`
- It needs to work on both MacOS and Windows
- If you are expecting to see this come out as a v2 release the change also needs to be made to both the `master` (v2) and the `develop` (v3) branches, and it will be a slightly different change as the temp code has changed to allow for macos stuff.

**Comment by tcsidevtools** [comment]
Thank you for the response. I have already completed most of the work that needs to be done but i will go back and ensure it meets those specifications.

I will also point out that on the v3 branch (`develop`) there is a breaking change which might have been unintentional. It is in `Squirrel.CommandLine/NugetConsole.cs` line 60. Previously release notes were not required to do a build.

```
            return CreatePackageFromMetadata(tempDir, command.PackDirectory.FullName, command.PackId, command.PackTitle,
                command.PackAuthors, command.PackVersion, command.ReleaseNotes.FullName, command.IncludePdb, libFolderName);
```
It looks like were accessing ReleaseNotes.FullName which, if no release notes were provided, ReleaseNotes would be null causing an unhandled exception. In my case we don't use the release notes feature and so I had to change it to ` command.ReleaseNotes?.FullName ?? ""` for it to allow me to build on this branch. I've no idea if that would have a deeper impact or if the requirements have changed for release notes so I can remove that from the PR before I post it if this change was intentional.

I will also say it looks like the Releases folder is not being generated anymore automatically and will cause an error if it doesnt exist. I'm not sure if that is also an intended change.
